docs: add extra description to Limitless Connection Plugin parameters (#1294)
aws/aws-advanced-jdbc-wrapper: docs/using-the-jdbc-driver/using-plugins/UsingTheLimitlessConnectionPlugin.md

|----------------------------------------------------|:-------:|:--------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------|---------------|

> | `limitlessTransactionRouterMonitorIntervalMs`      | Integer |    No    | This property is the interval in milliseconds, that the plugin polls the database for available transaction routers and their load metrics. A lower value will increase the frequency of polling, and a higher value will decrease the frequency of polling. <br><br>The default value of 15 seconds was chosen to match the expected refresh rate of load metric metadata from the database.<br><br>Note that there will always be a delay between when the database updates its load metric info and when the Limitless Connection Plugin polls for it. If your Limitless database experiences fluctuating load between transaction routers, you may want to consider lowering `limitlessTransactionRouterMonitorIntervalMs` to reduce this delay and ensure the Limitless Connection Plugin load balancing has fresher info to work with.                                                                                                                                                    | `15000`       | `30000`       |

Reviewer: Thought we discussed using 7 seconds ?

Reviewer: Other than that it looks good

Author: I plan on making the change to the default value in a separate branch/commit

> | `limitlessTransactionRouterMonitorDisposalTimeMs`  | Integer |    No    | This property is the time in milliseconds that if a Limitless transaction router monitor unused for, it will be disposed. This ensures that in times of long inactivity, the database isn't being needlessly polled and the resources associated with the monitor can be cleaned up. Note that when a new connection is created, a new Limitless transaction router monitor will also be created to resume polling the database.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | `600000`      | `300000`      |

Reviewer 2: ```suggestion
| `limitlessTransactionRouterMonitorDisposalTimeMs`  | Integer |    No    | This property is the time in milliseconds that a Limitless transaction router monitor can remain unused before it is disposed. This ensures that in times of long inactivity, the database isn't being needlessly polled and the resources associated with the monitor can be cleaned up. Note that when a new connection is created, a new Limitless transaction router monitor will also be created to resume polling the database.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | `600000`      | `300000`      |
```

> | `limitlessConnectMaxRetries`                       | Integer |    No    | This property is the max number of retries the Limitless Connection Plugin will attempt when trying to connecting to the database and it fails. During these retries, the plugin will attempt to connect to the least loaded transaction router that is known to be available. If the max number of connection retries is exceeded, then the plugin will throw a `SQLException`. In this scenario, it is likely that the database is in an unhealthy state, and the `SQLException` should be caught and handled by your application.                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | `5`           | `13`          |

Reviewer 2: ```suggestion
| `limitlessConnectMaxRetries`                       | Integer |    No    | This property is the max number of retries the Limitless Connection Plugin will attempt when failing to connect to the database. During these retries, the plugin will attempt to connect to the least loaded transaction router that is known to be available. If the max number of connection retries is exceeded, then the plugin will throw a `SQLException`. In this scenario, it is likely that the database is in an unhealthy state, and the `SQLException` should be caught and handled by your application.                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | `5`           | `13`          |
```

> | `limitlessWaitForTransactionRouterInfo`            | Boolean |    No    | In scenarios such as application start-up, the cache of available transaction routers and their load metric info, will be empty. If `limitlessWaitForTransactionRouterInfo` is set to `true`, the plugin will wait until cache is populated before selecting a transaction router and connecting to it. This may be beneficial for applications that create a large number of connections on start-up, so that these connections can then be load-balanced. <br><br>If this property set to `false` on application start-up, the plugin will not wait for the cache to be populated and default to using the DB Shard Group endpoint to connect to until there is available transaction router info. This will result connections being routed to a transaction router via Route 53 until the cache is populated. This may be beneficial for applications that prioritize quicker start-up times at the expense of some early connections not being load-balanced by the Limitless Connection Plugin. | `true`        | `false`       |

Reviewer 2: ```suggestion
| `limitlessWaitForTransactionRouterInfo`            | Boolean |    No    | In scenarios such as application start-up, the cache of available transaction routers and their load metric info will be empty. If `limitlessWaitForTransactionRouterInfo` is set to `true`, the plugin will wait until cache is populated before selecting a transaction router and connecting to it. This may be beneficial for applications that create a large number of connections on start-up, so that these connections can then be load-balanced. <br><br>If this property set to `false` on application start-up, the plugin will not wait for the cache to be populated and default to using the DB Shard Group endpoint to connect to until there is available transaction router info. This will result connections being routed to a transaction router via Route 53 until the cache is populated. This may be beneficial for applications that prioritize quicker start-up times at the expense of some early connections not being load-balanced by the Limitless Connection Plugin. | `true`        | `false`       |
```

| `limitlessGetTransactionRouterInfoMaxRetries`      | Integer |    No    | This property is the max number of times the Limitless Connection Plugin will retry fetching available transaction routers and their load metrics. These retries will occur if no transaction router information is returned. If this max is reached, a `SQLException` will be thrown by the AWS JDBC Driver. In this scenario, it is likely that the database is in an unhealthy state, and the thrown `SQLException` should be caught and handled by your application. <br><br>If your application prioritizes failing fast, then consider a lower value for this property. However, if your application prioritizes durability, then consider a higher value value this property.                                                                                                                                                                                                                                                                                                                  | `5`           | `10`          |

> | `limitlessGetTransactionRouterInfoRetryIntervalMs` | Integer |    No    | This property is the interval in milliseconds, between retries of fetching available transaction routers and their load metrics. <br><br>If your application prioritizes failing fast, then consider a lower value for this property. However, if your application prioritizes durability, then consider a higher value value this property.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | `300`         | `1000`        |

Reviewer 2: ```suggestion
| `limitlessGetTransactionRouterInfoRetryIntervalMs` | Integer |    No    | This property is the interval in milliseconds between retries of fetching available transaction routers and their load metrics. <br><br>If your application prioritizes failing fast, then consider a lower value for this property. However, if your application prioritizes durability, then consider a higher value value this property.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | `300`         | `1000`        |
```
